Monadfail

JackKelly-Bellroy opened the pull request:
Closes #61.
This PR adds the `MonadFail` instance to the `Marshal` monad, so that it works with base-4.13.0.0 and above. I haven't updated the version numbers, and leave it to you to cut a new release.

endgame commented:
And wearing my non-work, Hackage Trustee hat:

This means that the `base` version bounds on your previous releases are too loose (should be `< 4.13.0.0` instead of `< 5`), and metadata revisions should be published to Hackage for previous releases. I am able to assist with this, or even publish the revisions myself if you'd prefer.

JackKelly-Bellroy commented:
Annoyingly, we can't derive the MonadFail instances nor provide type signatures for the pattern synonyms if we want to continue supporting GHC 7.10.3. I have added a manual `Monad` and `MonadFail` instance with the necessary CPP, and removed the pattern synonyms. If you want to drop GHC 7.10.3 instead, I will rearrange this branch to give you a simpler PR.

mcfilib commented:
lgtm! i'll try and put together a release before the weekend.

endgame commented:
Great, thanks. Let me know if you'd like me to help out with the metadata revisions for previous hackage releases.

mcfilib commented:
go for it!

endgame commented:
Done for all versions `< 0.2`. I think the `base <5` on `0.2` is dubious but I'm not aware of other major changes coming down the line for `base`, so I left it alone.

P.S.: Thanks for dealing with this so quickly.
